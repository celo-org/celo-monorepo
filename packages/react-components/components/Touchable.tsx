import * as React from 'react'
import Touchable from 'react-native-platform-touchable'

export interface Props {
  borderless?: boolean
  onPress?: () => void
  children: React.ReactNode // must only have one direct child. see https://github.com/react-native-community/react-native-platform-touchable#touchable
  disabled?: boolean
  style?: any
  testID?: string
  hitSlop?: {
    top?: number
    left?: number
    bottom?: number
    right?: number
  }
}

export default class TouchableDefault extends React.PureComponent<Props> {
  delayedOnPress = () => {
    // Add a small delay so animations are seen
    setTimeout(this.props.onPress, 50)
  }
  effect = () => {
    return this.props.borderless
      ? Touchable.SelectableBackgroundBorderless()
      : Touchable.SelectableBackground()
  }
  render() {
    const props = { ...this.props }
    delete props.onPress
    return (
      <Touchable {...props} onPress={this.delayedOnPress} background={this.effect()}>
        {this.props.children}
      </Touchable>
    )
  }
}
