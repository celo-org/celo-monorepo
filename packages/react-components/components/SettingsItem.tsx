import Touchable from '@celo/react-components/components/Touchable'
import ForwardChevron from '@celo/react-components/icons/ForwardChevron'
import colors from '@celo/react-components/styles/colors'
import fontStyles from '@celo/react-components/styles/fonts'
import * as React from 'react'
import { StyleSheet, Text, View } from 'react-native'

export interface SettingsItemProps {
  testID?: string
  title?: string
  children?: React.ReactChild
  onPress?: () => void
}

class SettingsItem extends React.Component<SettingsItemProps> {
  render() {
    const { testID, title, onPress, children } = this.props

    return (
      <Touchable testID={testID} onPress={onPress}>
        <View style={style.settingsItemContainer}>
          <View style={[style.title]}>
            {children ? children : <Text style={fontStyles.body}>{title}</Text>}
          </View>
          <View style={[style.cta]}>
            <ForwardChevron height={16} />
          </View>
        </View>
      </Touchable>
    )
  }
}

const style = StyleSheet.create({
  settingsItemContainer: {
    marginLeft: 15,
    padding: 20,
    borderBottomWidth: 1,
    borderColor: colors.darkLightest,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    justifyContent: 'center',
  },
  cta: {
    justifyContent: 'center',
  },
})

export default SettingsItem
