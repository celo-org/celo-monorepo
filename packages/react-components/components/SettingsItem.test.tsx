import SettingsItem from '@celo/react-components/components/SettingsItem'
import Touchable from '@celo/react-components/components/Touchable'
import * as React from 'react'
import { fireEvent, render } from 'react-native-testing-library'

describe('SettingsItem', () => {
  describe('when pressed', () => {
    it('fires the onPress prop', () => {
      const handler = jest.fn()
      const { getByType } = render(<SettingsItem onPress={handler} title="SettingsItem" />)
      fireEvent.press(getByType(Touchable))
      expect(handler).toHaveBeenCalled()
    })
  })
  it('renders with minimum props', () => {
    const { getByText } = render(<SettingsItem onPress={jest.fn()} title="SettingsItem" />)
    expect(getByText('SettingsItem')).toBeTruthy()
  })
})
