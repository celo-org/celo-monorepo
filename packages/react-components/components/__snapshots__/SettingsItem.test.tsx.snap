// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsItemInput renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EDEEEF",
        "borderBottomWidth": 1,
        "marginLeft": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          title
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            Object {
              "flex": 0,
              "justifyContent": "flex-end",
              "paddingLeft": 16,
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[MockFunction]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "#2E3338",
                "flex": 1,
                "fontFamily": "Inter-Regular",
                "fontSize": 16,
                "lineHeight": 22,
                "paddingHorizontal": 0,
                "paddingVertical": 12,
              },
              undefined,
              Array [
                Object {
                  "color": "#9CA4A9",
                  "flex": 0,
                  "minWidth": 160,
                  "paddingVertical": 0,
                  "textAlign": "right",
                },
                Object {
                  "color": "#9CA4A9",
                },
              ],
            ]
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`SettingsItemSwitch renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EDEEEF",
        "borderBottomWidth": 1,
        "marginLeft": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          title
        </Text>
      </View>
      <AndroidSwitch
        accessibilityRole="switch"
        enabled={true}
        on={true}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      />
    </View>
  </View>
</View>
`;

exports[`SettingsItemTextValue renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EDEEEF",
        "borderBottomWidth": 1,
        "marginLeft": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          title
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#9CA4A9",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
              "marginRight": 8,
            }
          }
        >
          value
        </Text>
        <svg
          height={20}
          style={Object {}}
          viewBox="0 0 15 16"
          width={10}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.70376 9.12461L2.12123 15.7071C1.73071 16.0976 1.09755 16.0976 0.707015 15.7071C0.316497 15.3166 0.316497 14.6834 0.707015 14.2929L6.58575 8.41419L0.707091 2.53553C0.316573 2.14501 0.316573 1.51185 0.707091 1.12131C1.09761 0.730797 1.73079 0.730797 2.12131 1.12131L8.70708 7.70708C9.0976 8.09762 9.0976 8.73078 8.70708 9.1213L8.70376 9.12461Z"
            fill="#B4B9BD"
            style={Object {}}
          />
        </svg>
      </View>
    </View>
  </View>
</View>
`;

exports[`SettingsItemTextValue renders correctly without value 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#EDEEEF",
        "borderBottomWidth": 1,
        "marginLeft": 16,
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Inter-Regular",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          title
        </Text>
      </View>
    </View>
  </View>
</View>
`;
